File: spring_security/eg05_sb3_oauth2_resource_server_jwt_symmetric_key/src/main/java/com/sc/sb/auth/entity/ScUser.java
package com.sc.sb.auth.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Table(name = "SC_USER")
public class ScUser {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "ID")
    private Long id;

    @Column(name = "USER_NAME")
    private String userName;

    @Column(name = "USER_PASSWORD")
    private String userPassword;


    @ManyToMany
    @JoinTable(
            name = "SC_USER_ROLE",
            joinColumns = @JoinColumn(name="SC_USER_ID"),
            inverseJoinColumns = @JoinColumn(name = "SC_ROLE_ID"))
    private Set<ScRole> scRoles;
}

File: spring_security/eg06_kotlin_sb3_oauth2_resource_server_jwt_symmetric_key/src/main/kotlin/com/sc/sb/auth/entity/ScUser.kt
package com.sc.sb.auth.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany
import jakarta.persistence.Table

@Entity
@Table(name = "SC_USER")
data class ScUser(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        val id: Long,

        @Column(name = "USER_NAME")
        val userName: String,

        @Column(name = "USER_PASSWORD")
        val userPassword: String,

        @ManyToMany
        @JoinTable(
                name = "SC_USER_ROLE",
                joinColumns = [JoinColumn(name = "SC_USER_ID")],
                inverseJoinColumns = [JoinColumn(name = "SC_ROLE_ID")])
        private var scRoles: Set<ScRole?>?
)